File: csharp/Ice/value/PrinterI.cs
// **********************************************************************
//
// Copyright (c) 2003-2016 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

using Demo;

class PrinterI : Printer
{
    public override void printBackwards(Ice.Current current)
    {
        char[] arr = message.ToCharArray();
        for(int i = 0; i < arr.Length / 2; i++)
        {
            char tmp = arr[arr.Length - i - 1];
            arr[arr.Length - i - 1] = arr[i];
            arr[i] = tmp;
        }
        System.Console.Out.WriteLine(new string(arr));
    }
}

File: java/Manual/printer/PrinterI.java
// **********************************************************************
//
// Copyright (c) 2003-2016 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

public class PrinterI implements Demo.Printer
{
    @Override public void printString(String s, com.zeroc.Ice.Current current)
    {
        System.out.println(s);
    }
}
